### PR TITLE
Added new parameter comparisonAnalysis to the plugin

### PR DESCRIPTION
**Pull request by aseno-grammatech**
A new parameter named "`comparisonAnalysis`" has been introduced in order to add the capability of specifying a previous analysis to compare results with.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java [hunk 1]
```diff
@@ -386,80 +396,102 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnul
...
+        } else {
+            // Search for a previous, successful build that has the same ProjectName and hub URI.
+            //  We match hub URI since generally it is not reliable to compare results between different hubs.
+            //  Also, even if we want to consider other hubs, we may still be unable to communicate with them
+            //   due to CA certificate and user credential differences.
+            csLogger.writeInfo("Finding previous builds for comparison...");
+            Run<?,?> previousRun = run.getPreviousSuccessfulBuild();
+            while(previousRun != null 
+                && previousDataLoader == null
+            ) {
```
Comment (gt-dbechtold) [inline src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java:430]: Can we move this giant `else` clause containing the previous build search code to a dedicated method?  It seems complicated and procedural enough to deserve its own method.

File: src/main/resources/org/jenkinsci/plugins/codesonar/CodeSonarPublisher/help-baseAnalysis.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<div>
+    ID of a CodeSonar analysis to use as a base for "Warning Count Increase" conditions.
+    <br/>
+    If this field is not specified, the plugin will use the analysis results corresponding to the latest successful Jenkins build (if present).
+</div>
```
Comment (gt-dbechtold) [inline src/main/resources/org/jenkinsci/plugins/codesonar/CodeSonarPublisher/help-baseAnalysis.html:5]: We will need to add a description of this new parameter to the documentation before merging.

Comment (gt-dbechtold) [reply src/main/resources/org/jenkinsci/plugins/codesonar/CodeSonarPublisher/help-baseAnalysis.html:5]: @AmyGale please note the changes to the README.

Comment (AmyGale) [reply src/main/resources/org/jenkinsci/plugins/codesonar/CodeSonarPublisher/help-baseAnalysis.html:5]: I recommend calling this something else. "Base analysis" is established terminology in the context of incremental analysis.

Comment (gt-dbechtold) [reply src/main/resources/org/jenkinsci/plugins/codesonar/CodeSonarPublisher/help-baseAnalysis.html]: Changed to `comparisonAnalysis`

File: README.md [hunk 3]
```diff
@@ -353,16 +353,26 @@ analysis information from the hub.
...
       ```
       For example, to evaluate the results of the analysis whose ID is 81:
       ```
-      aid: 81,
+      aid: "81",
       ```
+      Note that the `aid` parameter is a string type, not a number type.
       In most cases the `projectFile` is the best way to identify the analysis of interest.
       However, you might want to use a different analysis in a more complex configuration involving multiple analyses.
+    * To specify the "previous" analysis to use as a base for comparing analysis results, add the following line after the `projectFile` line, where `<AID>` is the *analysis id* of you previous analysis.
```
Comment (AmyGale) [inline README.md:367]: you -> your

File: src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java [hunk 4]
```diff
@@ -455,28 +518,11 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnul
                 }
             }
         }
-
+
```
Comment (AmyGale) [inline src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java:521]: is this extra whitespace or something?

File: README.md (hunk 3 above)
Comment (gt-dbechtold) [reply README.md]: fixed, thanks

File: src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java (hunk 4 above)
Comment (gt-dbechtold) [reply src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java:521]: no, but I removed it to avoid extra stuff in diff.

Comment (gt-dbechtold) [reply src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java:521]: Err, well, maybe I removed the wrong thing and my local diff program doesn't fuss over whitespace so I didn't notice it until now.  Either way, it doesn't matter.

**Pull request merged**